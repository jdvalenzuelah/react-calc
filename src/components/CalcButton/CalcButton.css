.button {
    width: 75px;
    height: 75px;
    text-align: center;
    color: white;
    font-size: 40px;
    font-weight: 10;
    border: 0.031em solid black;
}

.button:disabled {
    color: black;
}

.operation {
    background-color: #FF9F0C;
}

.operation-helper {
    background-color: #48484A;
}

.number {
    background-color: #636565;
}

button:active.number,button:active.operation-helper {
    background-color: grey;
}

button:active.operation {
    background-color: rgba(255, 158, 12, 0.678);
}