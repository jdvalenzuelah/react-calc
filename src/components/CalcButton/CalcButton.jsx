import React from 'react';
import PropTypes from 'prop-types';
import './CalcButton.css';

const CalcButton = (props) => {
  const { label, type, disabled } = props;
  const classes = `button ${type}`;
  const handler = (e) => {
    props.hook(e.target.attributes.name.value);
  };
  return (
    <button className={classes} name={label} onClick={handler} disabled={disabled} type="button">
      {label}
    </button>
  );
};

CalcButton.propTypes = {
  label: PropTypes.string.isRequired,
  type: PropTypes.string,
  hook: PropTypes.func.isRequired,
  disabled: PropTypes.bool,
};

CalcButton.defaultProps = {
  type: 'number',
  disabled: false,
};
export default CalcButton;
