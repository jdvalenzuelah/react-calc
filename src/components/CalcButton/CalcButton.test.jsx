/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render } from '@testing-library/react';
import CalcButton from './CalcButton';

describe('CalcButton render tests', () => {
  test('Test if component is render correctly', () => {
    const container = render(<CalcButton label="Button" hook={() => null} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
