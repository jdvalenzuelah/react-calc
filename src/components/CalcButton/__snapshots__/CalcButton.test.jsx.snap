// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalcButton render tests Test if component is render correctly 1`] = `undefined`;
